ubccr/xdmod
# Attempt to provide drilldown filtering for group bys

## Description
This is a rough draft of providing the ability to filter out specific group bys ( via configuration file ) in OpenXDMoD while keeping the behavior the same in XSEDE. 

## Motivation and Context
We provide the ability to filter group_bys included as query descripters from the Usage tab / Metric Explorer Tab via roles.json->roles->role->query_descripters->query_descripter->show/hide. Where as the ability to filter a group_by from drop down lists is only accessible from within the GroupBy*.php code `[get|set]AvailableOnDrilldown`. 

These changes allow us to provide a 'visible' property for group_by records in datawarehouse.json that, in OpenXDMoD, when false will keep the group_by from appearing in the drill down list. 

## Tests performed
Manual Tests: 

**Test 1:**
- Login to OpenXDMOD
- Click the 'Usage' Tab
- Ensure that 'Jobs by Service Provider' is not visible 
- Plot some data from Jobs
- Click a point
- Ensure that 'Service Provider' is not included in the 'Drilldown to:' menu. 

**Test 2:**
- Login to XSEDE XDMoD
- Click the 'Usage' Tab
- Ensure that 'Jobs by Service Provider' is visible 
- Plot some data from Jobs
- Click a point
- Ensure that 'Service Provider' is included in the 'Drilldown to:' menu. 

## Types of changes


## Checklist:
